src/jit/copyprop.cpp

```diff
@@ -384,6 +525,246 @@ void Compiler::optBlockCopyProp(BasicBlock* block, LclNumToGenTreePtrStack* curS
     }
 }
 
+//------------------------------------------------------------------------------
+// optCopyPropFoldCopyBlks : Perform copy propagation for CopyBlk using currently live definitions on the
+//                           current block's variables.
+//
+// Arguments:
+//    block       -  Block the tree belongs to
+
+void Compiler::optCopyPropFoldCopyBlks(BasicBlock* block)
+{
+    // Propagate backwards, picking up:
+    //
+    // x1 (def) = x0 (last use)
+    // x2 (def) = x1 (last use)
```

Comment:
I don't see why `x0` and `x1` need to be last-use.
```
x1 = x0
x2 = x1
```
is
```
x1 = x0
x2 = x0
```
when `x1` is a local. Or at least a non volatile memory location.

Reply:
I'd agree; but I'm not very familiar with the Jit code so thought I'd start with a conservative change where I just removed variables that didn't serve much function (purely pass-through); then follow up where the copy is aliasing as I assumed it would be more involved?